agent/engine/ordering_integ_test.go

```diff
@@ -339,6 +340,74 @@ func TestDependencyHealthyTimeout(t *testing.T) {
 	waitFinished(t, finished, orderingTimeout)
 }
 
+
+func TestGranularStopTimeout(t *testing.T) {
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	stateChangeEvents := taskEngine.StateChangeEvents()
+
+	taskArn := "TestGranularStopTimeout"
+	testTask := createTestTask(taskArn)
+
+	parent := createTestContainerWithImageAndName(baseImageForOS, "parent")
+	dependency1 := createTestContainerWithImageAndName(baseImageForOS, "dependency1")
+	dependency2 := createTestContainerWithImageAndName(baseImageForOS, "dependency2")
+
+	parent.EntryPoint = &entryPointForOS
+	parent.Command = []string{"sleep 30"}
+	parent.Essential = true
+	parent.DependsOn = []apicontainer.DependsOn{
+		{
+			Container: "dependency1",
+			Condition: "START",
+		},
+		{
+			Container: "dependency2",
+			Condition: "START",
+		},
+	}
+
+	dependency1.EntryPoint = &entryPointForOS
+	dependency1.Command = []string{"trap 'echo caught' SIGTERM; while true; do echo hello; sleep 1s; done;"}
```

Comment:
can this Command be the same as the other one? Just so we can be confident that the timeout is the only changing parameter? 

This doesn't change the outcome of the test, it just more explicitly demonstrates what we have built